packages/trader/src/AppV2/Components/SymbolsSearchField/symbols-search-field.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useRef, useEffect, useState, forwardRef, Ref } from 'react';
+import { Button, SearchField } from '@deriv-com/quill-ui';
+import clsx from 'clsx';
+import { observer, useStore } from '@deriv/stores';
+import { Localize, localize } from '@deriv/translations';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { getContractTypesConfig } from '@deriv/shared';
+
+export type TSymbolsSearchField = {
+    isSearching: boolean;
+    setIsSearching: (input: boolean) => void;
+    searchValue: string;
+    setSearchValue: (input: string) => void;
+    marketCategoriesRef: React.MutableRefObject<HTMLDivElement | null>;
+};
+
+const SymbolsSearchField = observer(
+    ({ isSearching, setIsSearching, searchValue, setSearchValue, marketCategoriesRef }: TSymbolsSearchField) => {
+        const { ui } = useStore();
+        const { is_dark_mode_on } = ui;
+        const { contract_type } = useTraderStore();
+        const contract_name = getContractTypesConfig()[contract_type]?.title;
+        const inputRef = useRef<HTMLInputElement | null>(null);
+        const [prevScrollY, setPrevScrollY] = useState(marketCategoriesRef.current?.scrollTop);
+        const [isSearchFieldVisible, setIsSearchFieldVisible] = useState(true);
+
+        useEffect(() => {
+            const handleScroll = () => {
+                const currentScrollY = marketCategoriesRef.current?.scrollTop;
+
+                if (prevScrollY && currentScrollY !== null && currentScrollY !== undefined)
+                    setIsSearchFieldVisible(prevScrollY > currentScrollY);
+                setPrevScrollY(currentScrollY);
+            };
+            marketCategoriesRef.current?.addEventListener('scroll', handleScroll);
+            return () => {
+                marketCategoriesRef.current?.removeEventListener('scroll', handleScroll);
```

Comment:
Instead of passing `marketCategoriesRef`, adding/removing listeners directly to the DOM and handling the scroll logic in `useEffect` here, It's better to move `prevScrollY`/`setPrevScrollY` and `isSearchFieldVisible`/`setIsSearchFieldVisible` states to the parent (`ActiveSymbolsList`) and set the state in there using an `onScroll` callback which you can pass down to a node inside `MarketCategories` directly. This `SymbolsSearchField` would only use `prevScrollY` and `isSearchFieldVisible` props in this case.

Reply:
https://github.com/binary-com/deriv-app/pull/15776/commits/f3b63474f18510c998e3979c565e39287c74c1eb

previously i had the states in the parent but i moved it into this component to prevent rerender of the parent every time I scrolled

Reply:
In this case, if the re-render of the parent is costly, we need to memoize the other children of the parent that should not get re-rendered using `React.memo` + `React.useCallback` for callbacks. Please note that components that are wrapped into mobX's `observer()` are already memoized.

Reply:
i mean yeah thats why i elected to move the states down instead of memoizing everything